code/game/objects/items/devices/scanners.dm

```diff
@@ -298,27 +323,20 @@ SLIME SCANNER
 
 	to_chat(user, chat_box_healthscan(msgs.Join("<br>")))
 
-/obj/item/healthanalyzer/attack_self(mob/user)
-	mode = !mode
-	switch(mode)
-		if(1)
-			to_chat(user, "<span class='notice'>The scanner now shows specific limb damage.</span>")
-		if(0)
-			to_chat(user, "<span class='notice'>The scanner no longer shows limb damage.</span>")
-
 /obj/item/healthanalyzer/attackby(obj/item/I, mob/user, params)
-	if(istype(I, /obj/item/healthupgrade))
-		if(advanced)
-			to_chat(user, "<span class='notice'>An upgrade is already installed on [src].</span>")
-		else
-			if(user.unEquip(I))
-				to_chat(user, "<span class='notice'>You install the upgrade on [src].</span>")
-				add_overlay("advanced")
-				playsound(loc, I.usesound, 50, 1)
-				advanced = TRUE
-				qdel(I)
+	if(!istype(I, /obj/item/healthupgrade))
+		return ..()
+
+	if(advanced)
+		to_chat(user, "<span class='notice'>An upgrade is already installed on [src].</span>")
 		return
-	return ..()
+
+	user.unEquip(I)
```

Comment:
One last thing, You need to make sure they can actually unequip the upgrade, stuff like nodrop implants will prevent this

Reply:
Converted into a check.